src/main/java/org/jboss/tools/intellij/openshift/actions/component/DebugComponentAction.java

```diff
@@ -0,0 +1,161 @@
+
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public abstract class DebugComponentAction extends OdoAction {
+
+    private RunnerAndConfigurationSettings runSettings;
+
+    public DebugComponentAction() {
+        super(ComponentNode.class);
+    }
+
+    @Override
+    public boolean isVisible(Object selected) {
+        boolean visible = super.isVisible(selected);
+        if (visible) {
+            ComponentNode componentNode = (ComponentNode) selected;
+            Component component = (Component) componentNode.getUserObject();
+            return (isPushed(component) && isDebuggable(componentNode, component));
+        }
+        return false;
+    }
+
+    private boolean isDebuggable(ComponentNode componentNode, Component component) {
+        ApplicationNode applicationNode = (ApplicationNode) componentNode.getParent();
+        try {
+            Odo odo = ((ApplicationsRootNode) componentNode.getRoot()).getOdo();
+            LazyMutableTreeNode projectNode = (LazyMutableTreeNode) applicationNode.getParent();
+            ApplicationsRootNode root = (ApplicationsRootNode) applicationNode.getRoot();
+            ComponentInfo info = odo.getComponentInfo(root.getClient(), projectNode.toString(), applicationNode.toString(), component.getName());
+            return isDebuggable(info.getComponentTypeName());
+        } catch (IOException e) {
+            UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Debug"));
+        }
+        return false;
+    }
+
+    private boolean isPushed(Component component) {
+        return component.getState() == ComponentState.PUSHED;
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Odo odo) {
+        ComponentNode componentNode = (ComponentNode) selected;
+        Component component = (Component) componentNode.getUserObject();
+        ApplicationNode applicationNode = (ApplicationNode) componentNode.getParent();
+        LazyMutableTreeNode projectNode = (LazyMutableTreeNode) applicationNode.getParent();
+
+        Project project = anActionEvent.getData(CommonDataKeys.PROJECT);
+
+        RunManager runManager = RunManager.getInstance(project);
+        final Integer port = createOrFindPortFromConfiguration(runManager, component);
+        if (port == null) {
+            return;
+        }
+        ExecHelper.submit(() -> {
+            try {
+                // test if the port is available before running odo, if not, it means odo is already running. //TODO use 'odo debug status'
+                if (isAvailable(port)) {
+                    odo.debug(projectNode.toString(), applicationNode.toString(), component.getPath(), component.getName(), port);
+                    try {
+                        Thread.sleep(7000L); // TODO use again 'odo debug status' to wait until odo finish to start
```

Comment:
Do we really need to wait for 7s. Seems too long for users without a warning ?

Reply:
this is based on my local experiments, must depends on local environment.
i'll add a warning to inform user about this idle time.